platform/x11/os_x11.cpp

```diff
@@ -2849,11 +2849,19 @@ void OS_X11::set_context(int p_context) {
 	XClassHint *classHint = XAllocClassHint();
 	if (classHint) {
 
+		char *wm_class = (char *)"Godot";
 		if (p_context == CONTEXT_EDITOR)
 			classHint->res_name = (char *)"Godot_Editor";
 		if (p_context == CONTEXT_PROJECTMAN)
 			classHint->res_name = (char *)"Godot_ProjectList";
-		classHint->res_class = (char *)"Godot";
+
+		if (p_context == CONTEXT_ENGINE) {
+			classHint->res_name = (char *)"Godot_Engine";
+			wm_class = (char *)((String)GLOBAL_GET("application/config/name")).utf8().ptrw();
```

Comment:
Are all characters allowed for `wm_class`? `application/config/name` can have spaces, punctuation, etc.

Reply:
Yes. punctuations will be automatic escaped like I showed in the description.

Reply:
Another example:
![captura de tela de 2018-11-02 10-29-56](https://user-images.githubusercontent.com/1387165/47918106-5522d780-de8a-11e8-97cb-ae96ed59f7a4.png)
![captura de tela de 2018-11-02 10-30-09](https://user-images.githubusercontent.com/1387165/47918108-57853180-de8a-11e8-86df-864109a032c7.png)